bokeh/charts/_charts.py

```diff
@@ -144,17 +147,44 @@ def start_plot(self, xgrid, ygrid):
         if ygrid:
             self.make_grid(1, yaxis.ticker)
 
-        # Add tools
+        # Add tools if supposed to
         if self.tools:
+            # need to add tool to all underlying plots
             for plot in self._plots:
+                removed_tools = []
+                # only add tools if the underlying plot hasn't been customized
+                # by some user injection
                 if not plot.tools:
-                    if not self.categorical:
-                        pan = PanTool()
-                        wheelzoom = WheelZoomTool()
-                        reset = ResetTool()
-                        plot.add_tools(pan, wheelzoom, reset)
-                    previewsave = PreviewSaveTool()
-                    plot.add_tools(previewsave)
+                    tools = []
+                    # if tools have been specified as string let's reuse
+                    # plotting machinery to create the related tools
+                    if isinstance(self.tools, string_types):
+                        for tool in re.split(r"\s*,\s*", self.tools.strip()):
+                            # re.split will return empty strings; ignore them.
+                            if tool == "":
+                                continue
```

Comment:
ok... I was thinking in the bigger abstraction involving some other things... not only the tools-related, jeje... what you said seems ok... it could be added to plotting at first and then, when the time of the bigger abstraction comes, put it somewhere else (with bigger abstraction I am referring to things being used - or to be used - by different bokeh "zones", such as charts, widgets, plotting, embed, etc.